suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,194 @@
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+        });
+
+    const showAnotherEmptyAccountAlert = () =>
+        showAlert({
+            title: translate('moduleAddAccounts.alerts.anotherEmptyAccount.title'),
+            description: translate('moduleAddAccounts.alerts.anotherEmptyAccount.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate(
+                'moduleAddAccounts.alerts.anotherEmptyAccount.actionPrimary',
+            ),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+            secondaryButtonTitle: translate(
+                'moduleAddAccounts.alerts.anotherEmptyAccount.actionSecondary',
+            ),
+            onPressSecondaryButton: () => {
+                openLink(
+                    translate('moduleAddAccounts.alerts.anotherEmptyAccount.actionSecondaryUrl'),
+                );
+                hideAlert();
+            },
+        });
+
+    const addCoinAddress = ({ network, type }: { network: Network; type?: AccountType }) => {
+        if (!device?.state) {
+            return false;
+        }
+
+        const accountType = type ?? getDefaultAccountType({ network });
+
+        const currentAccountTypeAccounts = accounts.filter(
+            account => account.symbol === network.symbol && account.accountType === accountType,
+        );
+
+        // Do not allow adding more than 10 accounts of the same type
+        if (currentAccountTypeAccounts.length > LIMIT) {
+            showTooManyAccountsAlert();
+            return false;
+        }
+
+        // Do not allow adding another empty account if there is already one
+        const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
+
+        if (emptyAccounts.length > 0) {
+            showAnotherEmptyAccountAlert();
+            return false;
+        }
+
+        const addNetworkAccount = async (deviceState: string) => {
+            setIsAddingCoinAccount(true);
+            const account = await dispatch(
+                addNetworkAccountThunk({
+                    network,
+                    accountType,
+                    deviceState,
+                }),
+            ).unwrap();
+
+            setIsAddingCoinAccount(false);
+
+            if (account) {
+                navigation.navigate(RootStackRoutes.AccountDetail, {
+                    accountKey: account.key,
+                    tokenContract: undefined,
+                });
+            }
+        };
+
+        addNetworkAccount(device.state);
+    };
+
+    const onSelectedNetworkItem = (networkSymbol: NetworkSymbol) => {
+        if (isAddingCoinAccount) {
+            return;
+        }
+        const network = getNetwork({ networkSymbol });
+        if (network) {
+            addCoinAddress({ network });
+        } else {
```

Comment:
nit: I would remove this `else` branch, no added value from console.warn

Reply:
454699331c